nogo_main fail to build when to upgrade golang.org/x/tools

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of rules_go are you using?

<!-- Check io_bazel_rules_go in WORKSPACE if you're not sure -->

### What version of gazelle are you using?

<!-- Check bazel_gazelle in WORKSPACE if you're not sure -->

### What version of Bazel are you using?

<!-- Run "bazel version" to find out -->

Build label: 6.3.1-homebrew


### Does this issue reproduce with the latest releases of all the above?

if you update the tools like this

```
    go_repository(
        name = "org_golang_x_tools",
        build_file_proto_mode = "disable_global",
        importpath = "golang.org/x/tools",
        sum = "h1:ojD5zOW8+7dOGzdnNgersm8aPfcDjhMp12UfG93NIMc=",
        version = "v0.11.1",
    )
```

you will get the error

```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
external/io_bazel_rules_go/go/tools/builders/nogo_main.go:237:11: not enough arguments in call to pkg.facts.Encode
        have ()
        want (bool)
external/io_bazel_rules_go/go/tools/builders/nogo_main.go:399:54: not enough arguments in call to facts.NewDecoder(pkg.types).Decode
        have (func(pkg *types.Package) ([]byte, error))
        want (bool, func(pkgPath string) ([]byte, error))

```


### What operating system and processor architecture are you using?



### Any other potentially useful information about your toolchain?

<!-- C/C++ compiler, custom CROSSTOOL, remote execution? -->

### What did you do?

<!--
If possible, provide a minimal recipe for reproducing the error.
-->

### What did you expect to see?



### What did you see instead?

--- a/go/private/repositories.bzl
+++ b/go/private/repositories.bzl
@@ -82,6 +82,25 @@ def go_rules_dependencies(force = False):
         patch_args = ["-p1"],
     )
 
+    # Needed for go/tools/fetch_repo
+    # releaser:upgrade-dep golang tools
+    wrapper(
+        http_archive,
+        name = "org_golang_x_tools_go_vcs",
+        # v0.12.0, latest as of 2023-08-12
+        urls = [
+            "https://mirror.bazel.build/github.com/golang/tools/archive/refs/tags/go/vcs/v0.1.0-deprecated.zip",
+            "https://github.com/golang/tools/archive/refs/tags/go/vcs/v0.1.0-deprecated.zip",
+        ],
+        sha256 = "1b389268d126467105305ae4482df0189cc80a13aaab28d0946192b4ad0737a8",
+        strip_prefix = "tools-go-vcs-v0.1.0-deprecated/go/vcs",
+        patches = [
+            # releaser:patch-cmd gazelle -repo_root . -go_prefix golang.org/x/tools/go/vcs -go_naming_convention import_alias
+            Label("//third_party:org_golang_x_tools_go_vcs-gazelle.patch"),
+        ],
+        patch_args = ["-p1"],
+    )
+
     # releaser:upgrade-dep golang sys
     wrapper(
         http_archive,

--- a/go/tools/fetch_repo/BUILD.bazel
+++ b/go/tools/fetch_repo/BUILD.bazel
@@ -11,7 +11,7 @@ go_library(
     srcs = ["main.go"],
     importpath = "github.com/bazelbuild/rules_go/go/tools/fetch_repo",
     visibility = ["//visibility:private"],
-    deps = ["@org_golang_x_tools//go/vcs:go_default_library"],
+    deps = ["@org_golang_x_tools_go_vcs//:go_default_library"],
 )
 
 go_test(
@@ -19,5 +19,5 @@ go_test(
     size = "small",
     srcs = ["fetch_repo_test.go"],
     embed = [":go_default_library"],
-    deps = ["@org_golang_x_tools//go/vcs:go_default_library"],
+    deps = ["@org_golang_x_tools_go_vcs//:go_default_library"],
 )

